rfcs/xxxx-v5-database.md

```diff
@@ -0,0 +1,169 @@
+- Start Date: 2023-09-04
+- RFC PR: https://github.com/strapi/rfcs/pull/52
+# Summary
+
+We need to support development of new features. Most of them are going towards a more CMS like model. to support this approach we will introduce the notion of `documents` & `links`.
+
+# Motivation
+
+Bringing new features that were not possible in v4. The most important ones:
+
+## Draft and Publish
+
+Introducing a new version of draft & publish. One which allows having one draft and one published entry at the same time.
```

Comment:
I think it should be possible to have more than 2 publication states. Think about having diffrent environments where you want to publish a collection type.

Reply:
Could you elaborate more on the example? What would be the new publication states in those cases?

One additional feature we had in mind is private drafts, but that's something that might come after having the v5 foundations ready.

Reply:
Hi the setup in the case is like having multiple tenant and a collection that gets pubslihed to their different environments:

For example we have consultant that can have multiple web pages where they would be available. 

consultant
| id | name | publication_state | publication_id |
| 1 | Karl Heinz | draft | 0 |
| 2 | Karl Heinz | live| 1 |
| 3 | Karl Heinz | live:immobilien | 1 |
| 4 | Karl Heinz | live:beratung | 1 |

Other use case we have are emails that can be sent (by mail) and published to a webpage, were we pin the mail publication state, but the live publication can be changed later (for example because of typos)

mail
| id | text | publication_state | publication_id |
| 1 | xxx | draft | 0 |
| 2 | xxx | mail | 1 |
| 3 | xxx | live | 1 |